lively.collab/comments/commentMorph.js

```diff
@@ -70,31 +69,38 @@ export class CommentGroupMorph extends Morph {
   }
 
   getCommentMorphCount () {
-    return this.ui.commentMorphContainer.submorphs.length;
+    return this.commentMorphs.length;
   }
 
   getUnresolvedCommentMorphCount () {
-    return this.ui.commentMorphContainer.submorphs.filter((commentMorph) => !commentMorph.comment.isResolved()).length;
+    return this.commentMorphs.filter((commentMorph) => !commentMorph.comment.isResolved()).length;
   }
 
   async removeCommentMorph (comment) {
-    this.ui.commentMorphContainer.submorphs.forEach((commentMorph) => {
+    this.commentMorphs.forEach((commentMorph) => {
       if (commentMorph.comment.equals(comment)) {
-        commentMorph.remove();
+        commentMorph.delete();
+        remove(this.commentMorphs, commentMorph);
+        this.updateCommentContainerSubmorphs();
       }
     });
     this.updateCommentCountLabel();
   }
 
+  updateCommentContainerSubmorphs () {
+    if (this.isExpanded) {
+      this.ui.commentMorphContainer.submorphs = this.commentMorphs;
+    }
+  }
+
   applyExpanded () {
     Icon.setIcon(this.ui.collapseIndicator, this.isExpanded ? 'caret-down' : 'caret-right');
     if (!this.isExpanded) {
```

Comment:
```suggestion
    this.updateCommentContainerSubmorphs();
    // it should not be necessary to set extent manually, but layout doesn't change it automatically
    if (!this.isExpanded) this.ui.commentMorphContainer.extent = pt(0, 0);
```

Feels cleaner to me (no code duplication) and easier to remove once this bug was fixed